content/en/js3/blocks/promises/index.md

```diff
@@ -0,0 +1,67 @@
++++
+title = '🫱🏿‍🫲🏽 Promises'
+headless = true
+time = 15
+facilitation = false
+emoji= '🧩'
+[objectives]
+    1='Define a Promise'
+    2='Log a Promise to the console'
++++
+
+```mermaid
+graph LR
+    Asynchrony --> |delivered with| Promise{{🤝 Promises}}
+    Promise --> |resolve to a| Response{📤 Response}
+    Promise --> |join the| EventLoop{{Event Loop 🔁}}
+```
+
+To get data from a server, we make a request with `fetch`. We act on what comes back: the response. But what happens in the middle? We already know that JavaScript is single-threaded: it can only do one thing at a time.
+
+So do we just stop and wait? No! We have a special object to handle this time problem. Run this code in your terminal:
```

Comment:
Is this asking them to run this in `node` or in the web inspector console?

Reply:
Yes you can almost see me realising they haven't done try/catch in the text, can't you? 😂 

I had them running stuff in the browser but then I realised that has a variance we cannot control: CORS settings on the website apply to the console so if they open their console on MDN they will get different results to W3Schools or us. So then I moved it all to the REPL.  Let's specify REPL throughout